### ISSUE TITLE
`numprocs` configuration value won't be evaluated from environmental variable

### ISSUE THREAD

**Issue by psihonavt** [Supervisor/supervisor#750]
Looks like during reading configuration file supervisord treats value in `numprocs` as it is. eg:
```ini
[program:imap_poller]
command = bin/imap_poller_async
process_name=%(program_name)s_%(process_num)02d
numprocs=%(ENV_IMAP_FETCHERS_PROCESSES_NUM)s
killasgroup = true
autostart = false
startsecs = 3
stopwaitsecs = 30
redirect_stderr = true
stdout_logfile = %(here)s/../var/log/%(program_name)s_p%(process_num)s.stdout.log
autorestart = true
```
Running supervisord with such configuration will raise:
`Error: invalid literal for long() with base 10: '%(ENV_IMAP_FETCHERS_PROCESSES_NUM)s'`

Was it done by design or this is bug indeed?

**Comment by psihonavt** [comment]
Looks like it was fixed in newer versions of supervisor. We will upgrade ours and try again.

**Comment by mnaberez** [comment]
This was fixed in 2d6ca34582a8a07a5dd96ae45ef62cd58a459f4f, which was released in [Supervisor 3.2.0](https://github.com/Supervisor/supervisor/blob/3.2.0/CHANGES.txt#L11-L12).  Upgrade to 3.2.0 or later.

Minimal `supervisord.conf`:

    [supervisord]
    logfile = /tmp/supervisord.log
    
    [program:cat]
    command = /bin/cat
    process_name = %(program_name)s_%(process_num)s
    numprocs=%(ENV_CAT_NUMPROCS)s

Set `CAT_NUMPROCS` in the environment and start `supervisord` in the foreground with that config:

    $ supervisord --version
    3.2.0
    $ export CAT_NUMPROCS=3
    $ supervisord -n -c /path/to/supervisord.conf
    2016-04-29 06:57:59,008 INFO Increased RLIMIT_NOFILE limit to 1024
    2016-04-29 06:57:59,012 INFO supervisord started with pid 89156
    2016-04-29 06:58:00,016 INFO spawned: 'cat_0' with pid 89159
    2016-04-29 06:58:00,018 INFO spawned: 'cat_1' with pid 89160
    2016-04-29 06:58:00,020 INFO spawned: 'cat_2' with pid 89161
    2016-04-29 06:58:01,025 INFO success: cat_0 entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
    2016-04-29 06:58:01,026 INFO success: cat_1 entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
    2016-04-29 06:58:01,026 INFO success: cat_2 entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)

**Comment by jcrawford** [comment]
I am currently running Supervisor 3.2.0 and I cannot seem to get this to work properly.  I have tried several methods and did try the suggested solution above.

```
[program:laravel-worker]
process_name=%(program_name)s_%(process_num)02d
command=/usr/bin/php /var/www/html/artisan queue:work
autostart=true
autorestart=true
stdout_logfile=/var/log/worker.log
numprocs=%(ENV_WORKER_THREADS)s
```
in my `.env` file I have `WORKER_THREADS=1`. I even checked that it is in the shell

```
$ echo $WORKER_THREADS
2
```

When I try to run `supervisorctl reread` I get the following error.

```
ERROR: CANT_REREAD: Format string '%(ENV_WORKER_THREADS)s' for 'program:laravel-worker.numprocs' contains names ('ENV_WORKER_THREADS') which cannot be expanded. Available names: ENV_HOME, ENV_HOSTNAME, ENV_OLDPWD, ENV_PATH, ENV_PWD, ENV_SHLVL, ENV_TERM, ENV__, group_name, here, host_node_name, program_name in section 'program:laravel-worker' (file: '/etc/supervisor/conf.d/supervisor.conf')
```
I have tried using `%(WORKER_THREADS)s` and `%(ENV_WORKER_THREADS)s` and neither seems to work for me.  

Any thoughts?
